SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -0,0 +1,900 @@
++#define TDNF_CONF_KEY_EPOCH_TIME_LIMIT    "epochtimelimit"
+ #define TDNF_CONF_KEY_CLEAN_REQ_ON_REMOVE "clean_requirements_on_remove"
+ #define TDNF_CONF_KEY_REPODIR             "repodir" // typo, keep for back compatibility
+ #define TDNF_CONF_KEY_REPOSDIR            "reposdir"
+diff --git a/etc/bash_completion.d/tdnf-completion.bash b/etc/bash_completion.d/tdnf-completion.bash
+index 4e48040..285add2 100644
+--- a/etc/bash_completion.d/tdnf-completion.bash
++++ b/etc/bash_completion.d/tdnf-completion.bash
+@@ -92,7 +92,7 @@ _tdnf()
+ {
+     local c=0 cur __opts __cmds
+     COMPREPLY=()
+-    __opts="--assumeno --assumeyes --cacheonly --debugsolver --disableexcludes --disableplugin --disablerepo --downloaddir --downloadonly --enablerepo --enableplugin --exclude --installroot --noautoremove --nogpgcheck --noplugins --quiet --reboot --refresh --releasever --repo --repofrompath --repoid --rpmverbosity --security --sec --setopt --skip --skipconflicts --skipdigest --skipsignature --skipobsoletes --testonly --version --available --duplicates --extras --file --installed --whatdepends --whatrequires --whatenhances --whatobsoletes --whatprovides --whatrecommends --whatrequires --whatsuggests --whatsupplements --depends --enhances --list --obsoletes --provides --recommends --requires --requires --suggests --source --supplements --arch --delete --download --download --gpgcheck --metadata --newest --norepopath --source --urls"
++    __opts="--assumeno --assumeyes --cacheonly --debugsolver --disableexcludes --disableplugin --disablerepo --downloaddir --downloadonly --enablerepo --enableplugin --epochtimelimit --exclude --installroot --noautoremove --nogpgcheck --noplugins --quiet --reboot --refresh --releasever --repo --repofrompath --repoid --rpmverbosity --security --sec --setopt --skip --skipconflicts --skipdigest --skipsignature --skipobsoletes --testonly --version --available --duplicates --extras --file --installed --whatdepends --whatrequires --whatenhances --whatobsoletes --whatprovides --whatrecommends --whatrequires --whatsuggests --whatsupplements --depends --enhances --list --obsoletes --provides --recommends --requires --requires --suggests --source --supplements --arch --delete --download --download --gpgcheck --metadata --newest --norepopath --source --urls"
+     __cmds="autoerase autoremove check check-local check-update clean distro-sync downgrade erase help history info install list makecache mark provides whatprovides reinstall remove repolist repoquery reposync search update update-to updateinfo upgrade upgrade-to"
+     cur="${COMP_WORDS[COMP_CWORD]}"
+     _tdnf__process_if_prev_is_option && return 0
+diff --git a/include/tdnferror.h b/include/tdnferror.h
+index c9349a0..c207264 100644
+--- a/include/tdnferror.h
++++ b/include/tdnferror.h
+@@ -187,6 +187,15 @@ extern "C" {
+ #define ERROR_TDNF_HISTORY_ERROR 1801
+ #define ERROR_TDNF_HISTORY_NODB 1802
+ 
++#define ERROR_TDNF_FILTER_BASE 1900
++// filter MEMORY
```

Comment:
consider squeezing "TIME" between TDNF and FILTER or EPOCH or something to  allow for other future "FILTERS"

Reply:
added